lib/chef/knife/bootstrap.rb

```diff
@@ -549,7 +549,28 @@ def run
 
         $stdout.sync = true
         register_client
-        connect!
+        begin
+          connect!
+        rescue Train::Transports::SSHFailed => e
+          if e.message =~ /fingerprint (\S+) is unknown for "(.+)"/
+            fingerprint = $1
+            hostname,ip = $2.split(',')
+            puts "The authenticity of host '#{hostname} (#{ip})' can't be established."
+            # TODO: convert the SHA256 base64 value to hex with colons
+            # 'ssh' example output:
+            # RSA key fingerprint is e5:cb:c0:e2:21:3b:12:52:f8:ce:cb:00:24:e2:0c:92.
+            # ECDSA key fingerprint is 5d:67:61:08:a9:d7:01:fd:5e:ae:7e:09:40:ef:c0:3c.
+            puts "fingerprint is #{fingerprint}."
```

Comment:
recommend: #"#{hostname}'s fingerprint is: #{fingerprint}"

Reply:
I was trying to match the output of `ssh` which is two lines, the first line with the hostname is above the comments here. the second one is in the example comments, but to do it we need to convert the key from base64 at least in the case of an RSA SHA256 fingerprint, and I kinda punted on that.